app/src/main/java/com/provider/service/EntityConverterService.java

```diff
@@ -111,4 +113,44 @@ public ItemGetReturnModel convertItemToGetReturnModle(Item item) {
         modelMapper.map(item, ItemGetReturnModelResult.class);
     return new ItemGetReturnModel().ok(true).result(itemGetReturnModelResult);
   }
+
+  public List<TitleGetModel> convertObjectsListToTitleGetModel(List<Object[]> objects) {
+    List<TitleGetModel> titleGetModels = new ArrayList<>();
+    for (Object[] row : objects) {
+      Long id = ((Number) row[0]).longValue();
+      String title = (String) row[1];
+      String type = "";
+      Integer typeInteger = Integer.parseInt((String) row[2]);
+      switch (typeInteger) {
+        case 1:
+          type = "provider";
+          break;
+        case 2:
+          type = "item";
+          break;
+        case 3:
+          type = "sub item";
+          break;
+        default:
+          break;
+      }
+      String ref = "";
+      switch (type) {
+        case "provider":
+          ref = "/api/v1/provider/" + id;
+          break;
+        case "item":
+          ref = "/api/v1/item/" + id;
+          break;
+        case "sub item":
+          ref = "/api/v1/subitem" + id;
```

Comment:
according to REST for get sub item url: `/api/v1/provider/{provider-id}/item/{item-id}/subitem/{sub-item-id}`